File: sanidad-backend/src/main/java/com/farmacia/sanidadbackend/dto/VentaResponse.java
package com.farmacia.sanidadbackend.dto;

import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class VentaResponse {

    private Long id;
    private String numeroFactura;
    private LocalDateTime fecha;

    private Long clienteId;
    private String clienteNombre;
    private String clienteCedula;

    private Long usuarioId;
    private String usuarioUsername;

    private BigDecimal subtotal;
    private BigDecimal iva;
    private BigDecimal total;

    private String tipo;

    // 🔹 CAMPOS DE PAGO
    private BigDecimal montoUsadoSaldo;
    private BigDecimal montoEfectivo;
    private BigDecimal cambio;

    private List<VentaDetalleResponse> detalles;
}

File: SanidadMovil/app/src/main/java/com/sanidad/movil/data/remote/dto/VentaResponse.kt
package com.sanidad.movil.data.remote.dto

data class VentaResponse(
    val id: Long,
    val numeroFactura: String,
    val fecha: String, // ISO LocalDateTime
    val clienteId: Long?,
    val clienteNombre: String?,
    val clienteCedula: String?,
    val usuarioId: Long,
    val usuarioUsername: String,
    val subtotal: Double,
    val iva: Double,
    val total: Double,
    val tipo: String,
    val montoUsadoSaldo: Double,
    val montoEfectivo: Double,
    val cambio: Double,
    val detalles: List<VentaDetalleResponse>
)